src/Libraries/Text/Bounded.idr

```diff
@@ -2,63 +2,85 @@ module Libraries.Text.Bounded
 
 %default total
 
+
 public export
-record WithBounds ty where
-  constructor MkBounded
-  val : ty
-  isIrrelevant : Bool
+record Bounds where
+  constructor MkBounds
   startLine : Int
   startCol : Int
   endLine : Int
   endCol : Int
 
+export
+startBounds : Bounds -> (Int, Int)
+startBounds b = (b.startLine, b.startCol)
+
+export
+endBounds : Bounds -> (Int, Int)
+endBounds b = (b.endLine, b.endCol)
+
+export
+Eq Bounds where
+  (MkBounds sl sc el ec) == (MkBounds sl' sc' el' ec') =
+      sl == sl'
+   && sc == sc'
+   && el == el'
+   && ec == ec'
+
+public export
+record WithBounds ty where
+  constructor MkBounded
+  val : ty
+  isIrrelevant : Bool
+  bounds : Bounds
+
 export
 start : WithBounds ty -> (Int, Int)
-start b = (b.startLine, b.startCol)
+start = startBounds . bounds
 
 export
 end : WithBounds ty -> (Int, Int)
-end b = (b.endLine, b.endCol)
+end = endBounds . bounds
 
 export
 Eq ty => Eq (WithBounds ty) where
-  (MkBounded val ir sl sc el ec) == (MkBounded val' ir' sl' sc' el' ec') =
-    val == val' && ir == ir' && sl == sl' && sc == sc' && el == el' && ec == ec'
+  (MkBounded val ir bd) == (MkBounded val' ir' bd') =
+    val == val' && ir == ir' && bd == bd'
 
 export
 Show ty => Show (WithBounds ty) where
-  showPrec d (MkBounded val ir sl sc el ec) =
+  showPrec d (MkBounded val ir (MkBounds sl sc el ec)) =
     showCon d "MkBounded" (concat [showArg ir, showArg val, showArg sl, showArg sc, showArg el, showArg ec])
```

Comment:
Should probably introduce `Show Bounds` using `showCon "MkBounds" (...)`
and call that rather than unpacking the `Bounds` record here.